### PR TITLE
Refactor GHContentSearchBuilder to allow a Fork type

### PR DESCRIPTION
**Pull request by sahansera**
# Description

- Introduce a new enum `GHFork`.
- Refactor the existing code to use the new `GHFork` enum.
- Add a new unit test to test GitHub code search including results of forks.

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time.

- [x] Push your changes to a branch other than `main`. Create your PR from that branch.
- [x] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [x] Fill in the "Description" above.
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHFork.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package org.kohsuke.github;
+
+/**
+ * The enum for Fork search mode
+ */
+public enum GHFork {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHFork.java:6]: This is copied from https://github.com/hub4j/github-api/blob/bcb71a3c318d243c92b757310cf153b8afc983d5/src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java#L219-L253 right?

File: src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java [hunk 2]
```diff
@@ -76,8 +76,8 @@ public GHRepositorySearchBuilder forks(String v) {
      *      in forks</a>
      *
      */
-    public GHRepositorySearchBuilder fork(Fork fork) {
-        if (Fork.PARENT_ONLY.equals(fork)) {
+    public GHRepositorySearchBuilder fork(GHFork fork) {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java:79]: This is reasonable but to maintain backward compatibility we need to keep `Fork` and the related method and mark them both `@Deprecated`.

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Generally a good change.  Need adjustment maintain backward compatibility.

File: src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java (hunk 2 above)
Comment (sahansera) [reply src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java:113]: Thanks a lot for your feedback. I have updated PR with the requested changes. Please kindly have another look and kick-off the build checks when you have some time :)

File: src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java [hunk 3]
```diff
@@ -217,8 +251,9 @@ public GHRepositorySearchBuilder sort(Sort sort) {
     }
 
     /**
-     * The enum for Fork search mode
+     * The enum for Fork search mode. Note: Kept for backward compatibility. Use GHFork instead.
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java:254]:
```suggestion
     * The enum for Fork search mode.
     *
     * @deprecated Kept for backward compatibility. Use {@link GHFork} instead.
```

File: src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java [hunk 4]
```diff
@@ -47,9 +57,11 @@ public GHRepositorySearchBuilder size(String v) {
      * @param v
      *            the v
      * @return the gh repository search builder
+     * @deprecated use {@link #fork(GHFork)} instead.
      */
+    @Deprecated
     public GHRepositorySearchBuilder forks(String v) {
-        return q("forks:" + v);
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java:52]: Well, this is was a bug.

**Review by bitwiseman — APPROVED** [review]

File: src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java (hunk 4 above)
Comment (sahansera) [reply src/main/java/org/kohsuke/github/GHRepositorySearchBuilder.java:52]: Nice pickup! Thanks for reviewing this PR 🎉

**Pull request merged**